### DISCUSSION TITLE
Where to define categories: Desktop File or Appstream file?

### DISCUSSION THREAD

**Discussion by 1Maxnet1** [ximion/appstream#678] [General]
For context: I'm currently trying to upstream a bunch of metadata collected in the https://linuxphoneapps.org/ project to their respective app-repositories, so the metadata is can be updated/maintained where the app is maintained. Therefore we have a little script that compares what information is available in the upstream repository and what is available in our collection. When I create PRs/MRs to upstream the metainfo, one regurlarly upcoming question is:

> I already have defined the categories in the Desktop file. Why should I also define it in the metainfo.xml?

One answer here would be, because our little script only checks the metainfo.xml and not also takes into account the Desktop-File. But adding them only for our script seems not satisfying, because it will result in having defined the categories in two places, which may lead to issues when updating them in one place and forget to update them in another place. One of the benefits of having a metainfo.xml file in my opinion is that the metadata is defined only at one single place and not at different places. So What are your thoughts on that? Here are mine on the three options I see:

1. Keep them in desktop and metainfo.xml file: Manual work to keep them synchronized and hardly feasible if it is only for the little script checking the metadata.
2. Keep them only in the desktop file: I would need to adapt our checker (which is fine) but as far as I understand this would also result in the categories missing from the metadata collection that is e.g. used by Software Stores (like KDE Discover or GNOME Software) and therefore they wouldn't / could not show the Apps in the according categories.
3. Keep them only in the metainfo.xml and remove them from the Desktop file: This would be bad as well, as the Apps would not show up in the according category in a start menu. However, it might be possible to generate/complement the Desktop file at build or packaging stage. This would resolve the issue of having to maintain the data at two places, while ensuring to keep everything working.

I appreciate your input

**Comment by razzeee** [comment]
2.
> as far as I understand this would also result in the categories missing from the metadata collection that is e.g. used by Software Stores (like KDE Discover or GNOME Software) and therefore they wouldn't / could not show the Apps in the according categories.

I don't think that's correct, they will still be picked up by tooling and end up in the catalogue data (that's at least what happens with flatpak, you can't build successfully, without linking a `.desktop` file)
E.g. https://github.com/flathub/net.filebot.FileBot/blob/master/net.filebot.FileBot.desktop
Still works fine

3. I actually would be surprised, if that breaks things, but I haven't tested it (might not even have a DE, that shows that problem)



I usually have the categories etc in the `.desktop` file, but only cause I liked that more from a translation POV - which mostly means, I dislike `gettext` I guess.

**Comment by bbhtt** [comment]
I'd just keep them in desktop file and sync it with appstream by adding a `launchable` tag with `desktop-id`. That is, if these are desktop applications.

> I understand this would also result in the categories missing from the metadata collection that is e.g. used by Software Stores (like KDE Discover or GNOME Software) and therefore they wouldn't / could not show the Apps in the according categories.

Not if you have the launchable tag like above.

If it is a console application, you should have them in Metainfo only.

Note that, categories/keywords defined in Metainfo override the ones in the desktop file.
